AccessDeniedException in quarkus dev mode

### Describe the bug

I tried to upgrade to the newest version of quarkus. If I want to start it with `mvn clean compile quarkus:dev` I'm getting the error  java.nio.file.AccessDeniedException: /home/xyz/projects/my-project/database. This database folder has the rights `drwx------` as root, and that should not be changed.

Thanks in advance!

### Expected behavior

If I have not accessible folders outside `src`-folder should not end in access denied if I want to start quarkus in dev mode. If that is the case, there should be a possible exclude of folders/files.

### Actual behavior

Full stacktrace:
Caused by: java.io.UncheckedIOException: java.nio.file.AccessDeniedException: /home/xyz/projects/my-project/database
	at io.quarkus.devui.deployment.menu.WorkspaceProcessor.locateWorkspaceItems(WorkspaceProcessor.java:95)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.nio.file.AccessDeniedException: /home/vagrant/projects/chronapp/core-backend/database
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:90)
	at java.base/sun.nio.fs.UnixException.asIOException(UnixException.java:115)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newDirectoryStream(UnixFileSystemProvider.java:501)
	at java.base/java.nio.file.Files.newDirectoryStream(Files.java:482)
	at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:301)
	at java.base/java.nio.file.FileTreeWalker.next(FileTreeWalker.java:374)
	at java.base/java.nioiles.walkFileTree(Files.java:2833)
	at java.base/java.nio.file.Files.walkFileTree(Files.java:2870)
	at io.quarkus.devui.deployment.menu.WorkspaceProcessor.locateWorkspaceItems(WorkspaceProcessor.java:70)
	... 10 more

### How to Reproduce?

1. Creating a quarkus project with dev ui and version 3.23.2.
2. Create a folder with permissions `drwx------` as root at the same directory level as `pom.xml`.
3. Starting quarkus with `mvn clean compile quarkus:dev`

--- a/extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/WorkspaceProcessor.java
+++ b/extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/WorkspaceProcessor.java
@@ -3,6 +3,7 @@ package io.quarkus.devui.deployment.menu;
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.net.URI;
+import java.nio.file.AccessDeniedException;
 import java.nio.file.FileVisitResult;
 import java.nio.file.Files;
 import java.nio.file.Path;
@@ -74,7 +75,8 @@ public class WorkspaceProcessor {
                 Files.walkFileTree(projectRoot, new SimpleFileVisitor<Path>() {
                     @Override
                     public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                        if (Files.isHidden(dir) || ignoreFolders.contains(dir.getFileName().toString())) {
+                        if (Files.isHidden(dir) || ignoreFolders.contains(dir.getFileName().toString())
+                                || !Files.isReadable(dir) || !Files.isExecutable(dir)) {
                             return FileVisitResult.SKIP_SUBTREE;
                         }
                         return FileVisitResult.CONTINUE;
@@ -84,7 +86,7 @@ public class WorkspaceProcessor {
                     public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
                         String fileName = file.getFileName().toString();
                         boolean shouldIgnore = Files.isHidden(file)
-                                || file.startsWith(outputDir)
+                                || file.startsWith(outputDir) || !Files.isReadable(file) || !Files.isExecutable(file)
                                 || ignoreFilePatterns.stream().anyMatch(p -> p.matcher(fileName).matches());
 
                         if (!shouldIgnore) {
@@ -92,6 +94,18 @@ public class WorkspaceProcessor {
                             workspaceItems.add(new WorkspaceBuildItem.WorkspaceItem(name, file));
                         }
                         return FileVisitResult.CONTINUE;
+                    }
+
+                    @Override
+                    public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+                        if (exc instanceof AccessDeniedException) {
+                            if (Files.isDirectory(file)) {
+                                return FileVisitResult.SKIP_SUBTREE;
+                            } else {
+                                return FileVisitResult.CONTINUE;
+                            }
+                        }
+                        return super.visitFileFailed(file, exc);
                     }
                 });
 

